pkg/cluster/member.go

```diff
@@ -45,13 +60,24 @@ func (c *Cluster) updateMembers(known etcdutil.MemberSet) error {
 			c.memberCounter = ct + 1
 		}
 
-		members[name] = &etcdutil.Member{
+		member := &etcdutil.Member{
 			Name:         name,
 			Namespace:    c.cluster.Namespace,
 			ID:           m.ID,
 			SecurePeer:   c.isSecurePeer(),
 			SecureClient: c.isSecureClient(),
 		}
+
+		if c.cluster.Spec.SelfHosted != nil {
+			podIP, ok := peerPodIPs[m.Name]
```

Comment:
Don't list pod and create another map here.
Use the PeerURLs from ListMember response: https://github.com/coreos/etcd/blob/0f4a535c2fb7a2920e13e2e19b9eaf6b2e9285e5/etcdserver/etcdserverpb/rpc.pb.go#L1379-L1388

Reply:
Ah okay, good idea. That's in the format of `name1=podIP,name2=podIP` right? 

So split by `,` then match by member name

Reply:
It's the PeerURLs given to the etcd process.
For example, `PeerURLs[0]=http://10.1.0.1:2380`

Reply:
Assume only one PeerURL.
Also only need to handle self-hosted case.

Reply:
For 3 member clusters, there will be a slice of strings, e.g. `[]string{"http://10.1.0.1:2380","http://10.1.0.2:2380","http://10.1.0.3:2380"}`. 

How does the update method know which URL belongs to which member? One option is for member-0000 to have index 0, member-0001 have index 1, etc. Is it safe to assume the slice is always ordered?

Reply:
I don't catch your question :(
So let me be more detailed about what changes should be made here:

```Go
peerURL := m.PeerURLs[0] // "http://10.1.0.1:2380"
podIP := getIP(peerURL) // "10.1.0.1"
member.PodIP = podIP
```

Reply:
can you give me an example of `resp.Members`?

Is it like:

```
{Name: "member1", PeerURLs: []string{"http://10.1.0.1:2380","http://10.1.0.2:2380","http://10.1.0.3:2380"},
{Name: "member2", PeerURLs: []string{"http://10.1.0.1:2380","http://10.1.0.2:2380","http://10.1.0.3:2380"},
{Name: "member3", PeerURLs: []string{"http://10.1.0.1:2380","http://10.1.0.2:2380","http://10.1.0.3:2380"},
```

or:

```
{Name: "member1", PeerURLs: []string{"http://10.1.0.1:2380"},
{Name: "member2", PeerURLs: []string{"http://10.1.0.2:2380"},
{Name: "member3", PeerURLs: []string{"http://10.1.0.3:2380"},
```

?

Reply:
The second.
It's the PeerURLs that the member exposes.

Reply:
Ah okay, my mistake! I'll fix this tomorrow :)